### ISSUE TITLE
[Feature Request] Use Handheld Devices to "Connect" as a Wii U Gamepad

### ISSUE THREAD

**Issue by ThisNekoGuy** [cemu-project/Cemu#922]
Was thinking to myself it would be neat that, if you're playing a Wii U game on a desktop/laptop, that you could connect a secondary touchscreen device (such as a Steam Deck) as a window for Wii U gamepad-screen functionality.
Don't know if Cemu already supports with actual Wii U gamepads (I haven't tried) but having this would actually be a neat workaround for not strictly requiring the original gamepad hardware (or a second window cluttering the screen) to do so.

**Comment by Squall-Leonhart** [comment]
> I don't know if Cemu already supports with actual Wii U gamepads

There isn't any way to connect the gamepad to a pc, other than an obscure linux driver that requires a specific wireless dongle.

As to your idea,

https://www.reddit.com/r/cemu/comments/ut1k8w/guide_to_cemu_gamepad_with_steam_deck_or_android/

**Comment by ThisNekoGuy** [comment]
Most of that reddit post either mentions using Windows software or SteamLink (for Linux), and while those are technically solutions...
neither of those are "clean" (built-in)...

Would it not be feasible to forward the screen over a port on the network?

**Comment by Squall-Leonhart** [comment]
> Would it not be feasible to just forward the screen over a configurable port over the local network in-app?

No, the host still needs to construct a window/surface unique from the TV screen, so the host still needs to be able to project that window to a desktop surface to send it to the output device.

**Comment by ThisNekoGuy** [comment]
A 2nd Cemu installation could handle the host windowing but that would limit the solution to just handheld-PC x86-64 devices and Linux-powered tablets

**Comment by ThisNekoGuy** [comment]
Just realized that this kind of functionality may also be important for games that force the gamepad on a player when there's more than one local player (such as Hyrule Warriors)

**Comment by spudpiggy** [comment]
I saw this and I agree that it's a good idea. Just cus you can do it already doesn't mean it's 
@Squall-Leonhart I heavily doubt that a separate window is the only way to get cemu to even output gamepad video - even if it is, can't the host just keep it minimised?
The receiver can be its own app, available for mobile devices as well as handheld PCs (like the steam deck)

**Comment by Papermanzero** [comment]
try https://github.com/H-M-H/Weylus
You have a second screen (mobile or tablet) with touch input. Weylus need just some optimisations concerning latency but would be then the perfect solution for all multi screen emulations (DS, GC-GBA, 3DS, WiiU, Dreamcast etc)

**Comment by Wontell** [comment]
I would like to Bump this, we have Legion GO, Rog Ally, ayaneo KUN and many other device that can act as WiiU gamepad
Using such handheld as WiiU gamepad will be enable ultimate emualtion 1:1, even in future, 10-50 years in the future when these old consoles will be forgotten, if this is implement anyone with PC handheld will get to experience WIIU

This can be done trough WiFi, handheld connected to same router as the PC, as program runs on th handheld and connects to Cemu on PC

**Comment by Papermanzero** [comment]
It is not that easy. As said. You can do it right bow using weylus. Connect weylus to your tablet or phone. Take the second window and drop it to your tablet and phone. However the latency is still too high. A point-to-point connection would reduce the latency however this must be implemented in weylus. Wifi direct would be a solution for that.
